Document point index commands (create, drop and list) (#1259)
neo4j/neo4j-documentation: cypher/cypher-docs/src/docs/dev/ql/administration/indexes-for-search-performance/index.asciidoc

Once an index has been created, it will be managed and kept up to date by the DBMS.
Neo4j will automatically pick up and start using the index once it has been created and brought online.

There are multiple index types available: b-tree, fulltext, lookup, point, range, and text index types.
See <<administration-indexes-fulltext-search>> for more information about fulltext indexes.
Token lookup indexes contain nodes with one or more labels or relationship types, without regard for any properties.

Cypher enables the creation of b-tree and range indexes on one or more properties for all nodes or relationships with a given label or relationship type:

* An index that is created on a single property for any given label or relationship type is called a _single-property index_.
* An index that is created on more than one property for any given label or relationship type is called a _composite index_.

Differences in the usage patterns between composite and single-property indexes are described in <<administration-indexes-single-vs-composite-index>>.

> Additionally, text and point indexes are a form of single-property indexes, with the limitation that they only recognizes properties with string and point values, respectively.

Reviewer: Is it bad to call this a limitation in the docs? We could say something like - that supports queries specifically designed for text or point values, respectively, and therefor only index properties values of those types. Or maybe that is messier, don't know

Author: Compared to btree/range it is a limitation, but I can see if I can add something about it why

Reviewer 2: I think we use the word limitation in similar ways in other parts of the docs, so I do not think it is really an issue, So unless @Hunterness or @AlexicaWright might know a better way to phrase this, I think it is ok to leave as is

Reviewer 3: I don't think it is negative to call it a limitation in the docs.

> Additionally, text and point indexes are a form of single-property indexes, with the limitation that they only recognizes properties with string and point values, respectively.

Reviewer 3: ```suggestion
Additionally, text and point indexes are a kind of single-property indexes, with the limitation that they only recognize properties with string and point values, respectively.
```

> Nodes with the indexed label where the indexed property is of another type will not be included in the index.

Reviewer: Are people smart enough to understand that this goes for relationships as well if it is a relationship index?

Author: we should add it, thanks for pointing it out :)

> Nodes with the indexed label where the indexed property is of another type will not be included in the index.

Reviewer 3: ```suggestion
Nodes with the indexed label where the indexed property is of another type are not included in the index.
```

Differences in the usage patterns between composite and single-property indexes are described in <<administration-indexes-single-vs-composite-index>>.

Text and point indexes are also a version of single-property indexes, with the limitation that they only knows about properties with specific values.

> For text indexes these are string values and for point indexes these are point values. Other values in the indexed property will be ignored.

Reviewer 2: I think this could be formulated more clearly. Something more like the below perhaps:

Text and point indexes are also versions of single-property indexes, with the limitation that they only know about properties of the types string and point, respectively. Nodes with the indexed label where the indexed property is of another type will not be included in the index.

| Create a single-property index on nodes.

Index provider and configuration can be specified using the `OPTIONS` clause.

> .6+.^| Best practice is to give the index a name when it is created.

Author: remember to update the intro about index types

Index provider and configuration can be specified using the `OPTIONS` clause.

| [source, cypher, role=noplay]
----
CREATE RANGE INDEX [index_name] [IF NOT EXISTS]
FOR (n:LabelName)
ON (n.propertyName_1[,
    n.propertyName_2,
    …
    n.propertyName_n])
[OPTIONS "{" option: value[, ...] "}"]
----
| Create a range index on nodes, either on a single property or composite.

> Index provider can be specified using the `OPTIONS` clause.

Reviewer: Should we really mention index provider here?, we only support the range provider and there is no reason (at least not now) to ever send it in?
Question goes for all the new index types

Reviewer: Talked with Anton, decided that we should have it since it will be visible in createStatements mm.

Author: Since you can add it and see it in the show commands it makes sense to document it :)

Index provider can be specified using the `OPTIONS` clause.

| [source, cypher, role=noplay]
----
CREATE TEXT INDEX [index_name] [IF NOT EXISTS]
FOR (n:LabelName)
ON (n.propertyName)
[OPTIONS "{" option: value[, ...] "}"]
----
| Create a text index on nodes.

Index provider can be specified using the `OPTIONS` clause.

| [source, cypher, role=noplay]
----
CREATE TEXT INDEX [index_name] [IF NOT EXISTS]
FOR ()-"["r:TYPE_NAME"]"-()
ON (r.propertyName)
[OPTIONS "{" option: value[, ...] "}"]
----
| Create a text index on relationships.

Index provider can be specified using the `OPTIONS` clause.

| [source, cypher, role=noplay]
----
CREATE POINT INDEX [index_name] [IF NOT EXISTS]
FOR (n:LabelName)
ON (n.propertyName)
[OPTIONS "{" option: value[, ...] "}"]
----

> | Create a point index on nodes.

Author: Should we mention somewhere that point index only handles properties with point values? And if so, where would be good to add that?

Reviewer 2: Sounds like a good idea, see my answer on the text index PR.

Index provider can be specified using the `OPTIONS` clause.

| [source, cypher, role=noplay]
----
CREATE TEXT INDEX [index_name] [IF NOT EXISTS]
FOR (n:LabelName)
ON (n.propertyName)
[OPTIONS "{" option: value[, ...] "}"]
----

> | Create a text index on nodes.

Reviewer: Should we be super clear and mention here too that the index will only index property values of that type? Don't know how much repetition we usually do

Author: might be good yes

== Creating indexes

include::../indexes/create-a-single-property-b-tree-index-for-nodes.asciidoc[leveloffset=+1]

include::../indexes/create-a-single-property-b-tree-index-for-relationships.asciidoc[leveloffset=+1]

include::../indexes/create-a-single-property-b-tree-index-only-if-it-does-not-already-exist.asciidoc[leveloffset=+1]

include::../indexes/create-a-single-property-b-tree-index-with-specified-index-provider.asciidoc[leveloffset=+1]

include::../indexes/create-a-single-property-b-tree-index-with-specified-index-configuration.asciidoc[leveloffset=+1]

include::../indexes/create-a-composite-b-tree-index-for-nodes.asciidoc[leveloffset=+1]

include::../indexes/create-a-composite-b-tree-index-for-relationships.asciidoc[leveloffset=+1]

include::../indexes/create-a-composite-b-tree-index-with-specified-index-provider-and-configuration.asciidoc[leveloffset=+1]

include::../indexes/create-a-single-property-range-index-for-nodes.asciidoc[leveloffset=+1]

include::../indexes/create-a-single-property-range-index-for-relationships.asciidoc[leveloffset=+1]

include::../indexes/create-a-range-index-only-if-it-does-not-already-exist.asciidoc[leveloffset=+1]

> include::../indexes/create-a-range-index-specifying-the-index-provider.asciidoc[leveloffset=+1]

Reviewer: So here comes the question about index providers again. Is this something we should advertise or should we just skip these examples until we introduce any new index provider? Or do we need to have it here if anyone ever gets the idea of having their custom index provider? @burqen you wanna have an opinion on this one?

Reviewer: Keep, as seen in createStatements mm
